akka-discovery/src/main/scala/akka/discovery/ServiceDiscovery.scala

```diff
@@ -206,7 +206,27 @@ case object Lookup {
 
   private val SrvQuery = """^_(.+?)\._(.+?)\.(.+?)$""".r
 
-  private val DomainName = "^((?!-)[A-Za-z0-9-]{1,63}(?<!-)\\.)+[A-Za-z]{2,6}$".r
+  /**
+   * Validates domain name:
+   * - a node name has 1 to 63 chars
+   * - valid chars for a node name are: a-z, A-Z, 0-9 and -
+   * - a node name can't start with - character
```

Comment:
the RFC also seems to disallow starting names with numbers ("The labels must follow the rules for ARPANET host names.  They must
start with a letter, end with a letter or digit, and have as interior
characters only letters, digits, and hyphen"), should we follow that?

Reply:
I missed this one. We may be looking to an out-dated spec, because in the meantime I think this is not the case anymore. 

For instance, https://www.47deg.com/ is a valid domain.

Reply:
Here is the ARPANET RFC: https://tools.ietf.org/html/rfc819
It is stating the same, can't start with a digit, but document is from 1982.

Reply:
Hmm, perhaps we should not be so strict about hyphens either then... (e.g. https://stackoverflow.com/questions/2180465/can-domain-name-subdomains-have-an-underscore-in-it)

Reply:
Yeah, maybe it doesn't make sense to be restrictive at all. 

When we create an A/AAA lookup using only the name passed by the user, we don't check if it's a valid domain name. We just take it and run the lookup. 

We could argue that we should do the same for SRV lookup. 

```scala
val whatever = "123-name-"

Lookup(whatever) // valid A/AAA lookup

Lookup(whatever)
  .withPortName("http")
  .withProtocol("tpc") // 'valid' SRV lookup

Lookup.parseSrv(s"_http._tcp.$whatever") // invalid SRV lookup
```

The only reason I see for being strict is that we are 'parsing' something and therefore we kind of feel we should abide by some rules/format.

But we could also say that the parsing is only intended to split the string into something that we can use to lookup. Split into three parts: `portname`, `portocol` and `service-name`, but there is no restriction on any of the parts.

It's just a facility to go from:
`_http._tcp.some-service` 
to:
`Lookup("some-service").withPortName("http").withProtocol("tpc")`

Reply:
@raboof, I pushed some more changes making it compliant with ARPANET spec. 

I think this is a good enough approach. It's better to be too restrictive and relax later than the other way around. 

The previous implementation, which this PR is fixing, was not too restrictive, but it was wrong. It forced users to have at least one subdomain.